### PR TITLE
streamdeck-ui: 3.1.0 -> 4.1.2

### PR DESCRIPTION
**Pull request by ulrikstrid**
## Description of changes

For reviewers: Review the first commit since that is the actual change. The other commit is just the result of running `nix run .\#nixfmt-rfc-style pkgs/applications/misc/streamdeck-ui/default.nix`. Let me know if you want me to revert that commit.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/streamdeck-ui/default.nix [hunk 1]
```diff
@@ -74,25 +94,32 @@ python3Packages.buildPythonApplication rec {
...
+    [ xvfb-run ]
+    ++ (with python3Packages; [
+      pytest
+      pytest-qt
+      pytest-mock
+    ]);
 
   checkPhase = ''
+    # The tests needs to find the log file
+    export STREAMDECK_UI_LOG_FILE=$(pwd)/.streamdeck_ui.log
```
Comment (ulrikstrid) [inline pkgs/applications/misc/streamdeck-ui/default.nix:122]: Is there a better way of doing this?

**Comment by ulrikstrid** [comment]
Result of `nixpkgs-review pr 318899` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>streamdeck-ui</li>
    <li>streamdeck-ui.dist</li>
  </ul>
</details>

**Comment by ulrikstrid** [comment]
Been running this today and it works well and even works on my StreamDeck+ (which was the reason I wanted to update it). The main hurdle now is to get a OK on the code.

File: pkgs/applications/misc/streamdeck-ui/update-pillow.patch [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
  [tool.poetry.dependencies]
- python = ">=3.8,<3.12"
- streamdeck = "^0.9.3"
--pillow = "^9.4.0"
-+pillow = "^10.0.0"
- pynput = "^1.7.6"
+ python = ">=3.8,<3.13"
+ streamdeck = "^0.9.5"
+-pillow = "10.2.0"
++pillow = "^10.2.0"
```
Comment (superherointj) [inline pkgs/applications/misc/streamdeck-ui/update-pillow.patch:10]: This patch is going to conflict continuously. Consider:
https://nixos.org/manual/nixpkgs/stable/#using-pythonrelaxdepshook

File: pkgs/applications/misc/streamdeck-ui/default.nix [hunk 3]
```diff
@@ -1,69 +1,89 @@
...
   pname = "streamdeck-ui";
-  version = "3.1.0";
+  version = "4.1.2";
 
   src = fetchFromGitHub {
     repo = "streamdeck-linux-gui";
     owner = "streamdeck-linux-gui";
     rev = "v${version}";
-    sha256 = "sha256-AIE9j022L4WSlHBAu3TT5uE4Ilgk/jYSmU03K8Hs8xY=";
+    sha256 = "sha256-CSsFPGnKVQUCND6YOA9kfO41KS85C57YL9LcrWlQRKo=";
```
Comment (superherointj) [inline pkgs/applications/misc/streamdeck-ui/default.nix:32]: Use `hash`.

File: pkgs/applications/misc/streamdeck-ui/default.nix (hunk 1 above)
Comment (superherointj) [reply pkgs/applications/misc/streamdeck-ui/default.nix:122]: Can't answer.

File: pkgs/applications/misc/streamdeck-ui/default.nix [hunk 4]
```diff
@@ -74,33 +94,41 @@ python3Packages.buildPythonApplication rec {
...
+      setuptools
+      filetype
+      cairosvg
+      pillow
+      pynput
+      pyside6
+      streamdeck
+      xlib
+      importlib-metadata_6
+    ]
```
Comment (superherointj) [inline pkgs/applications/misc/streamdeck-ui/default.nix:109]: Would `evdev` dependency be missing?

https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/blob/main/pyproject.toml#L22C1-L22C6

https://pypi.org/project/evdev/

File: pkgs/applications/misc/streamdeck-ui/update-pillow.patch (hunk 2 above)
Comment (ulrikstrid) [reply pkgs/applications/misc/streamdeck-ui/update-pillow.patch]: Did not know about this, nice

File: pkgs/applications/misc/streamdeck-ui/default.nix (hunk 4 above)
Comment (ulrikstrid) [reply pkgs/applications/misc/streamdeck-ui/default.nix:108]: Added, thanks

**Comment by superherointj** [comment]
Package needs to be migrated to `by-name`.
From `pkgs/applications/misc/streamdeck-ui/default.nix` to `pkgs/by-name/st/streamdeck-ui/package.nix`.

**Comment by ulrikstrid** [comment]
> Package needs to be migrated to `by-name`. From `pkgs/applications/misc/streamdeck-ui/default.nix` to `pkgs/by-name/st/streamdeck-ui/package.nix`.

Is there anything other than moving it and changing all-packages.nix?

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -31670,7 +31670,7 @@ with pkgs;
 
   srain = callPackage ../applications/networking/irc/srain { };
 
-  streamdeck-ui = callPackage ../applications/misc/streamdeck-ui { };
+  streamdeck-ui = callPackage ../by-name/st/streamdeck-ui/package.nix { };
```
Comment (superherointj) [inline pkgs/top-level/all-packages.nix:31673]:
```suggestion
```

**Review by superherointj — APPROVED** [review]

**Pull request merged**